calvinpete/Glokeeper-backend
# #183532046 User should be able to filter & search

### What does this PR do?
This PR is to enable users to filter through properties, incomes, tenants and expenses

### Description of Task to be completed?
Looking for properties, incomes, tenants and expenses in their dozens can be daunting. Filters shall avail the members to create a search criterion of their own which helps them to narrow it down if they don’t know the title exactly..

### How should this be manually tested?
- Fetch this branch in your terminal by running the command `git fetch origin ft-filters-183532046`
- Check into the branch by running `git checkout ft-filters-183532046`
- Activate your virtual environment by running `source venv/bin/activate`
- Install dependencies by running `pip install -r requirements.txt`
- Migrate by running `python manage.py migrate`
- Run `python manage.py runserver`
- Filter the properties/incomes/tenants or expenses by providing the keywords
- To filter according to county, `use api/property/?county=Luzira`
- To filter according to merchant, use `api/property/hakim-apartments-kmhpp/expenses/?merchant=Safeboda`
- To filter and order results by property name, use `api/property/?county=Nakawa&ordering=-property_name`
- To search for a value within the article use `api/articles/?search=value_name`

## FILTER & ODERING
<img width="1005" alt="Screenshot 2022-10-17 at 05 24 52" src="https://user-images.githubusercontent.com/36506567/196076136-17767b10-8dae-4236-a71c-947d16f74479.png">

## SEARCH
<img width="1005" alt="Screenshot 2022-10-17 at 05 25 11" src="https://user-images.githubusercontent.com/36506567/196076183-22c0ced5-6473-474c-a95a-6db17721a901.png">

### What are the relevant pivotal tracker stories?
[#183532046](https://www.pivotaltracker.com/story/show/183532046)
